src/test/java/com/coveo/spillway/trigger/AbstractLimitTriggerTest.java

```diff
@@ -0,0 +1,64 @@
+package com.coveo.spillway.trigger;
+
+import static org.mockito.Mockito.*;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.coveo.spillway.limit.LimitDefinition;
+
+public class AbstractLimitTriggerTest {
+
+  private static final LimitDefinition LIMIT_DEFINITION =
+      new LimitDefinition("testLimit", 100, Duration.ofMinutes(1));
+
+  private class SimpleThresholdTrigger extends AbstractLimitTrigger {
+
+    public SimpleThresholdTrigger(LimitTriggerCallback callback) {
+      super(callback);
+    }
+
+    @Override
+    protected <T> boolean triggered(
+        T context, int currentLimitValue, LimitDefinition limitDefinition) {
+      return true;
+    }
+  }
+
+  private LimitTriggerCallback callback;
+  private AbstractLimitTrigger abstractLimitTrigger;
+
+  @Before
+  public void setup() {
+    callback = mock(LimitTriggerCallback.class);
+    abstractLimitTrigger = new SimpleThresholdTrigger(callback);
+  }
+
+  @Test
+  public void testCallbackNotTriggeredTwiceInSameBucket() {
+    Instant now = givenABucketStartingInstant();
+    abstractLimitTrigger.callbackIfRequired(null, 1, now, 1, LIMIT_DEFINITION);
+    abstractLimitTrigger.callbackIfRequired(null, 1, now.plusSeconds(10), 1, LIMIT_DEFINITION);
+    abstractLimitTrigger.callbackIfRequired(null, 1, now.plusSeconds(20), 1, LIMIT_DEFINITION);
+
+    verify(callback).trigger(any(LimitDefinition.class), any(Object.class));
```

Comment:
😱 Well damn. Either it changed or I'm crazy. I've always thought it was "at least once" by default. Thanks!